src/freenet/support/io/InetAddressIpv6FirstComparator.java

```diff
@@ -40,15 +42,15 @@ public int compare(InetAddress arg0, InetAddress arg1) {
 		Boolean reachable1 = reachabilityCache.get(b);
 		if (reachable0 == null) {
 			try {
-				reachable0 = arg0.isReachable(300);
+				reachable0 = arg0.isReachable((int) DEFAULT_MAX_PING_TIME);
```

Comment:
about the default max ping time: maybe needs to be tweaked, for example let's say I am connected to a VPN very far from my location, and I have 10x times more latency for the peers in my VPN (which are preferred, local LAN addresses) than public addresses closer to my country or within my country. Won't this make fred backoff my VPN peers and prefer my other peers because of lower latency?

Reply:
I’d push that to a later step if it is needed. Usually the default max ping time should be plenty for a real ping.